sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDiskStore.scala

```diff
@@ -102,22 +102,30 @@ class RapidsDiskStore(diskBlockManager: RapidsDiskBlockManager)
       stream: Cuda.Stream): (Long, Long, Long) = {
     incoming match {
       case fileWritable: RapidsBufferChannelWritable =>
+        val metricFn = incoming.storageTier match {
+          case StorageTier.DEVICE => GpuTaskMetrics.get.spillGpu2DiskTime[(Long, Long, Long)] _
+          case StorageTier.HOST => GpuTaskMetrics.get.spillHost2DiskTime[(Long, Long, Long)] _
```

Comment:
What's interesting is how much time and data was spilled to host or spilled to disk.  That's two metrics rather than the combinatorial setup here.  The origin of the data is less important and IMO not worth separating the metric out.

Reply:
@jlowe This PR deals only with time metrics, just to confirm: is your request a single spillToDiskTime metric?

Reply:
Yes, I think we're primarily only interested in the time spent transferring data to a particular destination, not the origin-destination combinations.